src/IO/Observer/WriteSimpleData.hpp

```diff
@@ -22,13 +22,21 @@ namespace observers {
 namespace ThreadedActions {
 
 /*!
- * \brief Append data to an h5::Dat subfile of `Tags::VolumeFileName`.
+ * \brief Append data to an h5::Dat subfile in an H5 file specified by
+ * `FilePrefixTag`.
  *
- * \details This is a streamlined interface for getting data to the volume file
- * associated with a node; it will simply write the `.dat` object
+ * \details This is a streamlined interface for outputting data on a node to an
+ * H5 file written by that node; it will simply write the `.dat` object
  * `subfile_name`, giving it the `file_legend` if it does not yet exist,
- * appending `data_row` to the end of the dat.
+ * appending `data_row` to the end of the dat. It will create the file if
+ * it doesn't already exist.
+ *
+ * \note The node number is appended to the filename, to ensure that the file
+ * written by each node is unique, even if `WriteSimpleData` is called on
+ * different nodes with the same `FilePrefixTag`. It is not safe to
+ * write to the same file from different nodes using `WriteSimpleData`.
```

Comment:
Just so I understand, even though CCE evolution is on a singleton, the output file name will still be something like `CharacteristicExtract3.h5` where 3 is the node CCE is on? But there won't be any files like `CharacteristicExtract0.h5`, `CharacteristicExtract1.h5`, etc...?

Reply:
That's right. I thought about removing that numerical suffix, but decided against it, because keeping it absolutely guarantees that the user can't attempt to write to the same file from two different nodes, which is unsafe.

Reply:
I like this decision.